### PR TITLE
Replace check pkt larger with gateway mtu

### PR DESCRIPTION
**Pull request by zshi-redhat**
This PR reverts 7e83226 and 78cf7be commits,
It removes the --disable-pkt-mtu-check flag and
adds --disable-gateway-mtu flag

Signed-off-by: Zenghui Shi zshi@redhat.com

### CODE REVIEW

**Comment by zshi-redhat** [comment]
/hold

hold this PR per comment: https://github.com/ovn-org/ovn-kubernetes/pull/2490#issuecomment-953120465

File: go-controller/pkg/ovn/gateway_test.go [hunk 1]
```diff
@@ -57,6 +57,9 @@ func generateGatewayInitExpectedNB(testData []libovsdb.TestData, expectedOVNClus
 		Name:     gwRouterPort,
 		MAC:      util.IPAddrToHWAddr(joinLRPIPs[0].IP).String(),
 		Networks: networks,
+		Options: map[string]string{
+			"gateway_mtu": fmt.Sprintf("%d", util.GetMaxFrameLength()),
```
Comment (dceara) [inline go-controller/pkg/ovn/gateway_test.go:61]: [A new feature was added](https://github.com/ovn-org/ovn/commit/acffcc0dde6c641aeaf516e0b600813c7bc4d489) to upstream OVN recently, `gateway_mtu_bypass`, we should probably use it to bypass check_pkt_larger checks for types of traffic that don't need it, for example for TCP.

File: go-controller/pkg/ovn/gateway_init.go [hunk 2]
```diff
@@ -115,17 +115,24 @@ func (oc *Controller) gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet,
 		gwLRPNetworks = append(gwLRPNetworks, gwLRPIfAddr.String())
 	}
 
+	var options map[string]string
+	if !config.Gateway.DisableGatewayMTU {
+		options = map[string]string{"gateway_mtu": fmt.Sprintf("%d", util.GetMaxFrameLength())}
```
Comment (dceara) [inline go-controller/pkg/ovn/gateway_init.go:120]: [A new feature was added](https://github.com/ovn-org/ovn/commit/acffcc0dde6c641aeaf516e0b600813c7bc4d489) to upstream OVN recently, `gateway_mtu_bypass`, we should probably use it to bypass check_pkt_larger checks for types of traffic that don't need it, for example for TCP.

Comment (zshi-redhat) [reply go-controller/pkg/ovn/gateway_init.go:120]: @trozet Do you think this new feature (gateway_mtu_bypass) would address partially [your concern](https://github.com/ovn-org/ovn-kubernetes/pull/2490#issuecomment-953120465) of no rate limiting for ovn-controller?

**Comment by trozet** [comment]
stale PR, please reopen and rebase if you want

**Pull request closed without merging**